src/Data/Enum.purs

```diff
@@ -37,10 +37,12 @@ derive newtype instance ordCardinality :: Ord (Cardinality a)
 -- | Type class for enumerations.
 -- |
 -- | Laws:
--- | - `succ a > pred a`
--- | - `pred a < succ a`
--- | - `pred >=> succ >=> pred = pred`
--- | - `succ >=> pred >=> succ = succ`
+-- | - Successor: `maybe true (a < _) (succ a)`
+-- | - Predecessor: `maybe true (_ < a) (pred a)`
+-- | - Succ retracts pred: `pred a >>= succ >>= pred = pred a`
+-- | - Pred retracts succ: `succ a >>= pred >>= succ = succ a`
+-- | - Non-skipping succ: `b <= a || maybe false (_ <= b) (succ a)`
```

Comment:
I don't know what I'd like to see in the specs, but for -quickcheck-laws, how about something like this? My beginner mind has less problems parsing it than the `maybe true`/`maybe false`/`any`/`all` alternatives.

```
    successor :: a -> Boolean
    successor a = forAllA $ (a < _) <$> (succ a)

    predecessor :: a -> Boolean
    predecessor a = forAllA $ (_ < a) <$> (pred a)

    forAllA :: Maybe Boolean -> Boolean
    forAllA = fromMaybe true

    nonSkippingSucc :: a -> a -> Boolean
    nonSkippingSucc a b = given (a < b) $ (_ <= b) <$> (succ a)

    nonSkippingPred :: a -> a -> Boolean
    nonSkippingPred a b = given (b < a) $ (b <= _) <$> (pred a)

    given :: Boolean -> Maybe Boolean -> Boolean
    given a b = not a || fromMaybe true b
```